runtime/doc/lua.txt

```diff
@@ -3670,4 +3670,87 @@ totable({f}, {...})                                       *vim.iter.totable()*
     Return: ~
         (table)
 
+
+==============================================================================
+Lua module: vim.snippet                                          *vim.snippet*
+
+active()                                                    *snippet.active()*
+    Returns `true` if there's an active snippet in the current buffer.
+
+    Return: ~
+        (boolean)
+
+cancel()                                                    *snippet.cancel()*
+    Cancels the current snippet.
+
+expand({input})                                             *snippet.expand()*
+    Expands the given snippet text.
+
+    Parameters: ~
+      • {input}  (string)
+
+jump({direction})                                             *snippet.jump()*
+    Jumps (if possible) within the active snippet in the given direction.
+
+    Parameters: ~
+      • {direction}  vim.snippet.Direction
+
+jumpable({direction})                                     *snippet.jumpable()*
+    @alias vim.snippet.Direction | -1 # Previous | 1 # Next Returns `true` if
+    there is an active snippet which can be jumped in the given direction.
+
+    Parameters: ~
+      • {direction}  vim.snippet.Direction
+
+    Return: ~
+        (boolean)
+
+new({index}, {bufnr}, {range})                                 *snippet.new()*
```

Comment:
@lewis6991 @clason @gpanders or anyone else that's familiar with the intricacies of the type generator: I just want to generate the documentation for the returned table in this module (`M`, which has the public snippet API). As noticed in this diff, `gen_vimdoc` is also including the other utility classes in this module (`Session` and `Tabstop`). I guess I can try just marking all of those methods with `@nodoc` to prevent them being included here, but that's cumbersome. Do you know of a better workaround with the existing doc infrastructure?

Reply:
I think `@nodoc` is The Way, at the moment.